inverse_covariance/clean.py

```diff
@@ -0,0 +1,323 @@
+                self.row_mean_ = .0
+                self.n_cols_seen_ = 0
+                if self.with_std:
+                    self.row_var_ = .0
+                else:
+                    self.row_var_ = None
+            self.row_mean_, self.row_var_, self.n_cols_seen_ = \
+                _incremental_mean_and_var(X.T, self.row_mean_,
+                                          self.row_var_,
+                                          self.n_cols_seen_
+                                          )
+
+        if self.with_std:
+            self.row_scale_ = _handle_zeros_in_scale(np.sqrt(self.row_var_))
+            self.col_scale_ = _handle_zeros_in_scale(np.sqrt(self.col_var_))
+        else:
+            self.row_scale_ = None
+            self.col_scale_ = None
+
+        return self
+
+    def transform(self, X, y='deprecated', copy=None):
+        """Perform standardization by centering and scaling
+        Parameters
+        ----------
+        X : array-like, shape [n_rows, n_cols]
+            The data used to scale along the features axis.
+        y : (ignored)
+            .. deprecated:: 0.19
+               This parameter will be removed in 0.21.
+        copy : bool, optional (default: None)
+            Copy the input X or not.
+        """
+        if not isinstance(y, string_types) or y != 'deprecated':
+            warnings.warn("The parameter y on transform() is " # noqa
+                          "deprecated since 0.19 and will be removed in 0.21",
+                          DeprecationWarning)
+
+        check_is_fitted(self, 'row_scale_')
+
+        copy = copy if copy is not None else self.copy
+        X = check_array(X, accept_sparse=None, copy=copy, warn_on_dtype=True,
+                        estimator=self, dtype=FLOAT_DTYPES)
+
+        if sparse.issparse(X):
+            print('Input is sparse')
+            raise NotImplemented(
+                "Algorithm for sparse matrices currently not supported.")
+        else:
+            X = twoway_standardize(X)
+        return X
+
+    def inverse_transform(self, X, copy=None):
+        """Scale back the data to the original representation
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data used to scale along the features axis.
+        copy : bool, optional (default: None)
+            Copy the input X or not.
+        Returns
+        -------
+        X_tr : array-like, shape [n_samples, n_features]
+            Transformed array.
+        """
+        check_is_fitted(self, 'row_scale_')
+
+        copy = copy if copy is not None else self.copy
+        if sparse.issparse(X):
+            print('Input is sparse')
+            raise NotImplementedError(
+                'Algorithm for sparse matrices currently not supported.')
+        else:
+            raise NotImplementedError(
+                'Two Way standardization not reversible with accuracy')
+            X = np.asarray(X)
```

Comment:
this code will never execute

Reply:
make sure there is a test that tests each of these methods, that would have picked this up

Reply:
Seems like this should be a warning not an exception.